packages/catalog-model/src/entity/policies/ReservedFieldsEntityPolicy.ts

```diff
@@ -17,47 +17,70 @@
 import { EntityPolicy } from '../../types';
 import { Entity } from '../Entity';
 
-const DEFAULT_RESERVED_ENTITY_FIELDS = [
-  'apiVersion',
-  'kind',
-  'uid',
-  'etag',
-  'generation',
-  'name',
-  'namespace',
-  'labels',
-  'annotations',
-  'spec',
-];
+type ExceptWhere = 'metadata' | 'spec' | 'labels' | 'annotations';
+type ReservedFields = Record<string, ExceptWhere[]>;
+
+const DEFAULT_RESERVED_ENTITY_FIELDS: ReservedFields = {
```

Comment:
Feel it would be nice to have this constant be a list of `.` separated names, i.e. `['apiVersion', 'kind', 'spec', 'metadata.uid']` etc., and then convert to whatever format is best for the runtime check.

Reply:
done